films/usecase/core.go

```diff
@@ -0,0 +1,257 @@
+package usecase
+
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/configs"
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/films/repository/crew"
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/films/repository/film"
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/films/repository/genre"
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/films/repository/profession"
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/pkg/models"
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/pkg/requests"
+)
+
+type ICore interface {
+	GetFilmsAndGenreTitle(genreId uint64, start uint64, end uint64) ([]models.FilmItem, string, error)
+	GetFilmInfo(filmId uint64) (*requests.FilmResponse, error)
+	GetActorInfo(actorId uint64) (*requests.ActorResponse, error)
+	GetActorsCareer(actorId uint64) ([]models.ProfessionItem, error)
+	GetGenre(genreId uint64) (string, error)
+	FindFilm(title string, dateFrom string, dateTo string,
+		ratingFrom float32, ratingTo float32, mpaa string, genres []string, actors []string) ([]models.FilmItem, error)
+	FavoriteFilms(userId uint64) ([]models.FilmItem, error)
+	FavoriteFilmsAdd(userId uint64, filmId uint64) error
+	FavoriteFilmsRemove(userId uint64, filmId uint64) error
+}
+
+type Core struct {
+	lg         *slog.Logger
+	films      film.IFilmsRepo
+	genres     genre.IGenreRepo
+	crew       crew.ICrewRepo
+	profession profession.IProfessionRepo
+}
+
+func GetCore(cfg_sql *configs.DbDsnCfg, lg *slog.Logger) (*Core, error) {
+	var films film.IFilmsRepo
+	var genres genre.IGenreRepo
+	var actors crew.ICrewRepo
+	var professions profession.IProfessionRepo
+	var err error
```

Comment:
Множественный var добавляли в язык не просто так - сложность чтения заметно снижается, если делать, например, так:

```suggestion
	var (
		films film.IFilmsRepo
		genres genre.IGenreRepo
		actors crew.ICrewRepo
		professions profession.IProfessionRepo
		err error
	)
```